File: docs/sphinx/examples/cpp/providers/photonics.cpp
// Compile and run with:
// ```
// nvq++ --target photonics photonics.cpp
// ./a.out
// ```

#include "cudaq/photonics.h"
#include "cudaq.h"

struct photonicsKernel {
  void operator()() __qpu__ {
    cudaq::qvector<3> qutrits(2);
    plus(qutrits[0]);
    plus(qutrits[1]);
    plus(qutrits[1]);
    mz(qutrits);
  }
};

int main() {

  auto counts = cudaq::sample(photonicsKernel{});
  for (auto &[k, v] : counts) {
    printf("Result / Count = %s : %lu\n", k.c_str(), v);
  }
}

File: docs/sphinx/examples/python/providers/photonics.py
import cudaq

cudaq.set_target("photonics-cpu")


@cudaq.kernel
def photonicsKernel():
    qumodes = [qudit(3) for _ in range(2)]
    create(qumodes[0])
    create(qumodes[1])
    create(qumodes[1])
    mz(qumodes)


counts = cudaq.sample(photonicsKernel)
print(counts)

state = cudaq.get_state(photonicsKernel)
print(state)
